site/content/entry/yuzu-mini-dev-2/index.md

```diff
@@ -0,0 +1,133 @@
+
+{{< single-title-imgs
+   "Cave Story+ (First yuzu build) - We've come a long way from that!"
+   "./first.png"
+>}}
+
+**Q: That's really interesting to hear. Before yuzu and Switch emulation, have you ever worked on other older emulators?**
+
+**R:** No, I've never worked on a non-Switch emulator. The closest thing I did related to emulation was a clone of Pico-8 Celeste for the NES in 6502 assembly.
+It was a fun experience because I learnt the curse and blessing that it is, working that close to the hardware.
+And it showed me what a pain it was, to program classics like Super Mario Bros. 3.
+
+
+**Q: Let's get a bit technical. What areas of switch emulation have you worked on mainly and why? Which other areas keep you interested?**
+
+**R:** I've worked on the graphics department and the compute department of Switch emulation.
+For those that don't know, the Switch's GPU - it's a Nvidia Tegra X1, the same architecture of a GTX 960+.
+Emulating a modern GPU (Tegra) by using other modern GPUs (user end), is my area of interest. 
+
+As for why I like it, it's because GPUs nowadays are like mini sub-computers inside a machine.
+The list of functionality present in the Tegra X1 grows bigger the more we look into it; since it’s a Nvidia device, it can execute CUDA kernels (programs that run some computations in the GPU, primarily used for scientific calculations or accelerated computations).
+It has the 5 traditional shader stages, an extra vertex shader stage, compute shaders, while at the same time it supports the supposedly deprecated features from the D3D9 era like rendering with quads and alpha testing.
+
+From time to time, I try to get into Core emulation, but it never ends up with good results. Funnily enough, while I was investigating the SSBU crash, I totally thought I made a breakthrough  - only to find out I was disassembling the wrong module. (LOL)
+
+
+**Q: What are some of the challenges you generally face, when working on GPU emulation?**
+
+**R:** One of biggest challenges, is the inherent variable state of Nvidia GPUs.
+Sometimes graphics APIs like OpenGL and Vulkan have requirements that are the "common ground", something that's needed by the lowest denominator.
+We have to write very annoying workarounds to get that functionality working in the emulator.
+
+<p style="color:cyan"><b><i>Here are some sneak peeks from my Vulkan test build. It's still missing a lot of things, but it's not too bad either.</b></i></p>
+
+{{< imgs
+   "./botw.png|The Legend of Zelda - Breath of the Wild"
+   "./onepiece.png|One Piece - Unlimited World"
+   "./smo.png|Super Mario Odyssey"
+>}}
+
+**Q: What motivates you every day to keep working on this stuff?**
+
+**R:** Emulating modern GPU features on hardware with those same features, is something that has only been possible in the recent times, due to the increasing similarities in hardware between consoles and desktops.
```

Comment:
Should be: `Emulating modern GPU features, on hardware with those same features, is something`